pkg/test/mgmtapi.go

```diff
@@ -0,0 +1,127 @@
+package test
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	"github.com/k8ssandra/cass-operator/pkg/httphelper"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/mocks"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/stargate"
+	"github.com/stretchr/testify/mock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ManagementApiFactoryAdapter func(
+	ctx context.Context,
+	datacenter *cassdcapi.CassandraDatacenter,
+	client client.Client,
+	logger logr.Logger) (cassandra.ManagementApiFacade, error)
+
+var defaultAdapater ManagementApiFactoryAdapter = func(
+	ctx context.Context,
+	datacenter *cassdcapi.CassandraDatacenter,
+	client client.Client,
+	logger logr.Logger) (cassandra.ManagementApiFacade, error) {
+
+	m := new(mocks.ManagementApiFacade)
+	m.On(EnsureKeyspaceReplication, mock.Anything, mock.Anything).Return(nil)
+	m.On(ListTables, stargate.AuthKeyspace).Return([]string{"token"}, nil)
+	m.On(CreateTable, mock.MatchedBy(func(def *httphelper.TableDefinition) bool {
+		return def.KeyspaceName == stargate.AuthKeyspace && def.TableName == stargate.AuthTable
+	})).Return(nil)
+	m.On(ListKeyspaces, "").Return([]string{}, nil)
+	m.On(GetSchemaVersions).Return(map[string][]string{"fake": {"test"}}, nil)
+	return m, nil
+}
+
+type FakeManagementApiFactory struct {
+	adapter ManagementApiFactoryAdapter
+}
+
+func (f *FakeManagementApiFactory) Reset() {
+	f.adapter = nil
+}
+
+func (f *FakeManagementApiFactory) SetAdapter(a ManagementApiFactoryAdapter) {
+	f.adapter = a
+}
+
+func (f *FakeManagementApiFactory) NewManagementApiFacade(
+	ctx context.Context,
+	dc *cassdcapi.CassandraDatacenter,
+	client client.Client,
+	logger logr.Logger) (cassandra.ManagementApiFacade, error) {
+
+	if f.adapter != nil {
+		return f.adapter(ctx, dc, client, logger)
+	}
+	return defaultAdapater(ctx, dc, client, logger)
+}
+
+type ManagementApiMethod string
+
+const (
+	EnsureKeyspaceReplication = "EnsureKeyspaceReplication"
+	GetKeyspaceReplication    = "GetKeyspaceReplication"
+	CreateKeyspaceIfNotExists = "CreateKeyspaceIfNotExists"
+	AlterKeyspace             = "AlterKeyspace"
+	ListKeyspaces             = "ListKeyspaces"
+	CreateTable               = "CreateTable"
+	ListTables                = "ListTables"
+	GetSchemaVersions         = "GetSchemaVersions"
+)
+
+type FakeManagementApiFacade struct {
+	*mocks.ManagementApiFacade
+}
+
+func NewFakeManagementApiFacade() *FakeManagementApiFacade {
+	return &FakeManagementApiFacade{ManagementApiFacade: new(mocks.ManagementApiFacade)}
+}
+
+func (f *FakeManagementApiFacade) GetLastCall(method ManagementApiMethod, args ...interface{}) int {
```

Comment:
Is this generated with Mockery? If so I'd suggest to:

* Move the generated code to the `mocks` package;
* Modify the `mocks` package README;
* Modify the `mocks` target in the Makefile.

Reply:
Hmm after reading this again, I think you wrote this code on top of the generated mock. Please disregard this comment.